src/Handler/Table/Create/CreateTableHandler.php

```diff
@@ -63,25 +67,63 @@ public function __invoke(
                 'nullable' => $column->getNullable(),
                 'default' => $column->getDefault() === '' ? null : $column->getDefault(),
             ]);
-            $columns[] = new BigqueryColumn($column->getName(), $columnDefinition);
+            $createTableOptions['schema']['fields'][] = SQLtoRestDatatypeConverter::convertColumnToRestFormat(
+                new BigqueryColumn($column->getName(), $columnDefinition)
+            );
+        }
+
+        $meta = MetaHelper::getMetaFromCommand(
+            $command,
+            CreateTableCommand\BigQueryTableMeta::class
+        );
+        if ($meta !== null) {
+            assert($meta instanceof CreateTableCommand\BigQueryTableMeta);
+            if ($meta->getTimePartitioning() !== null) {
+                $createTableOptions['timePartitioning'] = [
+                    'type' => $meta->getTimePartitioning()->getType(),
+                    'field' => $meta->getTimePartitioning()->getField(),
+                    'expirationMs' => $meta->getTimePartitioning()->getExpirationMs(),
+                ];
+            }
+            if ($meta->getClustering() !== null) {
+                $createTableOptions['clustering'] = [
+                    'fields' => ProtobufHelper::repeatedStringToArray($meta->getClustering()->getFields()),
+                ];
+            }
+            if ($meta->getRangePartitioning() !== null) {
+                assert($meta->getRangePartitioning()->getRange() !== null);
+                $createTableOptions['rangePartitioning'] = [
+                    'field' => $meta->getRangePartitioning()->getField(),
+                    'range' => [
+                        'start' => $meta->getRangePartitioning()->getRange()->getStart(),
+                        'end' => $meta->getRangePartitioning()->getRange()->getEnd(),
+                        'interval' => $meta->getRangePartitioning()->getRange()->getInterval(),
+                    ],
+                ];
+            }
+            $createTableOptions['requirePartitionFilter'] = $meta->getRequirePartitionFilter();
         }
-        $columnsCollection = new ColumnCollection($columns);
 
-        $builder = new BigqueryTableQueryBuilder();
         /** @var string $datasetName */
         $datasetName = $command->getPath()[0];
         $bqClient = $this->clientManager->getBigQueryClient($runtimeOptions->getRunId(), $credentials);
-        $dataset = $bqClient->dataset($datasetName);
-
-        $createTableSql = $builder->getCreateTableCommand(
-            $dataset->id(),
-            $command->getTableName(),
-            $columnsCollection,
-            [] // primary keys aren't supported in BQ
-        );
+        if ($runtimeOptions->getRunId() !== '') {
+            $createTableOptions['labels'] = ['run_id' => $runtimeOptions->getRunId(),];
```

Comment:
Přihodil sem run_id do labels tabulky, tak půjde najít ze kterého jobu ta tabulka pochází.

Reply:
Hmm, asi to nicomu nevadi no, mozno by nebolo odveci to pak tahat napr do telemetrie